core/src/main/java/com/linecorp/armeria/client/endpoint/SlowStartAwareEndpointSelectionStrategy.java

```diff
@@ -0,0 +1,167 @@
+            endpointsAndWeights = new EndpointsAndWeights(endpoints);
+            endpointGroup.addListener(this::scheduleUpdateRequestWeight);
+        }
+
+        @Override
+        public EndpointGroup group() {
+            return endpointGroup;
+        }
+
+        @Override
+        public EndpointSelectionStrategy strategy() {
+            return SlowStartAwareEndpointSelectionStrategy.this;
+        }
+
+        @Override
+        public Endpoint select(ClientRequestContext ctx) {
+            final int currentSequence = sequence.getAndIncrement();
+            return endpointsAndWeights.selectEndpoint(currentSequence);
+        }
+
+        private void scheduleUpdateRequestWeight(List<Endpoint> newEndpoints) {
+            if (updateEndpointWeightFuture != null && !updateEndpointWeightFuture.cancel(false)) {
+                updateEndpointWeightFuture = null;
+            }
+            newEndpoints.forEach(e -> currentWeights.put(e, new EndpointAndStep(e)));
+            updateEndpointWeightFuture =
+                    clientFactory.eventLoopGroup()
+                                 .scheduleAtFixedRate(() -> updateEndpointWeight(newEndpoints),
+                                                      0,
+                                                      slowStartInterval.toMillis(),
+                                                      TimeUnit.MILLISECONDS);
+        }
+
+        private void updateEndpointWeight(List<Endpoint> endpoints) {
+            endpointsAndWeights = new EndpointsAndWeights(
+                    endpoints.stream()
+                             .map(this::updateWeight)
+                             .collect(toImmutableList()));
+        }
+
+        private Endpoint updateWeight(Endpoint e) {
+            final EndpointAndStep endpointAndStep = currentWeights.get(e);
+            if (endpointAndStep == null) {
+                return e;
+            }
+            final int originalWeight = e.weight();
+            final int newWeight = endpointWeighter.compute(e, endpointAndStep.stepUp(), numberOfSteps);
+            if (originalWeight <= newWeight) {
+                currentWeights.remove(e);
+                return e;
+            }
+            return e.withWeight(newWeight);
+        }
+    }
+
+    private static final class EndpointAndStep {
+        private final Endpoint endpoint;
+        private int step;
+
+        private EndpointAndStep(Endpoint endpoint) {
+            this.endpoint = endpoint;
+        }
+
+        Endpoint endpoint() {
+            return endpoint;
+        }
+
+        int stepUp() {
+            return ++step;
+        }
+    }
+
+    /**
+     * Controls an {@link Endpoint} weight to ramp up a request load to the {@link Endpoint}.
+     */
+    @FunctionalInterface
+    public interface EndpointWeighter {
+        EndpointWeighter DEFAULT = (e, step, maxStep) -> (int) (e.weight() * (1.0 * step / maxStep * 100));
```

Comment:
- Shouldn’t this be `(double) e.weight() * step / maxStep`? Why do wee multiply by 100?
- `linear()`?